packages/contracts-bedrock/scripts/autogen/generate-semver-lock/main.go

```diff
@@ -0,0 +1,141 @@
+package main
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+func main() {
+	if err := run(); err != nil {
+		panic(err)
+	}
+}
+
+func run() error {
+	// find semver files
+	files, err := findSemverFiles()
```

Comment:
Honestly we can probably just throw all of this logic into a single `run()` function for now.